Sources/Build/Buildable.swift

```diff
@@ -9,20 +9,40 @@
 */
 
 import PackageType
+import struct Utility.Path
 
 protocol Buildable {
     var targetName: String { get }
     var isTest: Bool { get }
 }
 
+extension CModule {
+    ///Returns the build directory path of a CModule
+    func buildDirectory(prefix: String) -> String {
+        return Path.join(prefix, "\(c99name).build")
+    }
+}
+
 extension Module: Buildable {
     var isTest: Bool {
         return self is TestModule
     }
 
-    var Xcc: [String] {
+    func XccFlags(prefix: String) -> [String] {
         return recursiveDependencies.flatMap { module -> [String] in
-            if let module = module as? CModule {
+            if let module = module as? ClangModule {
+                ///For ClangModule we check if there is a user provided module map
+                ///otherwise we return with path of generated one.
+                ///We will fail before this is ever called if there is no module map.
+                ///FIXME: The user provided modulemap should be copied to build dir
+                ///but that requires copying the complete include dir because it'll
+                ///mostly likely contain relative paths.
+                if module.moduleMapPath.isFile {
+                    return ["-Xcc", "-fmodule-map-file=\(module.moduleMapPath)"]
+                }
+                let genModuleMap = Path.join(module.buildDirectory(prefix), module.moduleMap)
+                return ["-Xcc", "-fmodule-map-file=\(genModuleMap)"]
```

Comment:
I still think it is unfortunate that we are more or less recomputing state here which is independently computed by `generateModuleMap`. I don't think this is fixable in the current SwiftPM architecture, but it would be good to include a comment which points out that the behavior here is tied to that of the other function (and vice versa), so that if someone updates one they know to look at the other.